File: quicksort/recursive-sum/python/sum.py
# !/usr/bin/env python
# -*- coding: utf-8 -*-

def sum(list):
  if len(list) == 0:
    return 0
  else:
    return list[0] + sum(list[1:]) 

print sum([1, 3, 5, 4, 6, 5])    # 24

File: quicksort/recursive-sum/javascript/sum.js
function sum(list) {
  if (list.length == 0) {
    return 0;
  } else {
    var newArray = list.slice(1, list.length);
    return list[0] + sum(newArray);
  }
}

console.log(sum([1, 3, 5, 4, 6, 5]));    // 24

